channeldb/channel.go

```diff
@@ -729,7 +729,7 @@ type OpenChannel struct {
 	RevocationKeyLocator keychain.KeyLocator
 
 	// TODO(roasbeef): eww
-	Db *DB
+	Db *ChannelStateDB
```

Comment:
If all we need is a db backend here, can we pass `ChannelStateDB.db` to this `Db` field?

Reply:
Yeah, for the main code this might not be a big refactor and something we'll eventually want. But unfortunately the test code is relying on this to still be a `ChannelStateDB` instead of only the backend... I'd like to defer this more involved refactor to another PR.